src/block.js

```diff
@@ -71,15 +95,26 @@ Block.fromBuffer = function (buffer) {
 }
 
 Block.prototype.byteLength = function (headersOnly) {
-  if (headersOnly || !this.transactions) return 80
+  if (coins.isZcash(this.network)) {
```

Comment:
Almost half the code appears to be special casing for Zcash. Perhaps all of that should rather be separated into a method that parses Zcash blocks.

Reply:
I would do it if I could make it private, otherwise, having a `Block.prototype.byteLengthZcash` would pollute the methods namespace.

A better solution is to implement inheritance per supported coin. This would make things sooo much better. Unfortunatly, that's too much work for this diff